lib/star_rate.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:flutter/material.dart';
+
+class StarRate extends StatelessWidget {
+   const StarRate({
+    super.key,
+    required this.starColor,
+    required this.rate,
+  });
+
+  final double rate;
+  static const double maxStars = 5;
+  static const double rateTextSize = 20;
+  static const double paddingLeftBetweenStarsRate = 8.0;
+  static const double paddingLeftStars = 10;
+  static const double paddingTopBetweenImageStars = 12;
+
+  final Color starColor;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.only(left: paddingLeftStars, top: paddingTopBetweenImageStars),
+      child: Row(
+        children: [
+          for (int i = 0; i < maxStars; i++) starCreate(i, rate, starColor),
+          Padding(
+            padding: const EdgeInsets.only(left: paddingLeftBetweenStarsRate),
+            child: Text(
+              "$rate",
+              style: const TextStyle(
+                fontSize: rateTextSize,
+                color: Colors.white70,
+              ),
+            ),
+          )
+        ],
+      ),
+    );
+  }
+
+  starCreate(int i, double rate, Color starColor) {
```

Comment:
the build method should be the last in a class, move this one above it, and use a more descriptive name for the "i" parameter

Reply:
this one is not resolved Franco